satpy/writers/cf_writer.py

```diff
@@ -216,43 +505,97 @@ def _collect_datasets(self, datasets, kwargs):
+            epoch (str):
+                Reference time for encoding of time coordinates
+            flatten_attrs (bool):
+                If True, flatten dict-type attributes
+            exclude_attrs (list):
+                List of dataset attributes to be excluded
+            include_lonlats (bool):
+                Always include latitude and longitude coordinates, even for datasets with area definition
+            pretty (bool):
+                Don't modify coordinate names, if possible. Makes the file prettier, but possibly less consistent.
+        """
         logger.info('Saving datasets to NetCDF4/CF.')
-        # XXX: Should we combine the info of all datasets?
-        filename = filename or self.get_filename(**datasets[0].attrs)
-        datas, start_times, end_times = self._collect_datasets(datasets, kwargs)
 
-        dataset = xr.Dataset(datas)
-        try:
-            dataset['time_bnds'] = make_time_bounds(dataset,
-                                                    start_times,
-                                                    end_times)
-            dataset['time'].attrs['bounds'] = "time_bnds"
-        except KeyError:
-            logger.warning('No time dimension in datasets, skipping time bounds creation.')
-
-        header_attrs = kwargs.pop('header_attrs', None)
+        if groups is None:
+            # Write all datasets to the file root without creating a group
+            groups_ = {None: datasets}
+        else:
+            # User specified a group assignment using dataset names. Collect the corresponding datasets.
+            groups_ = defaultdict(list)
+            for dataset in datasets:
+                for group_name, group_members in groups.items():
+                    if dataset.attrs['name'] in group_members:
+                        groups_[group_name].append(dataset)
+                        break
+
+        # Write global attributes to file root (creates the file)
+        filename = filename or self.get_filename(**datasets[0].attrs)
 
+        root = xr.Dataset({}, attrs={'history': 'Created by pytroll/satpy on {}'.format(datetime.utcnow())})
         if header_attrs is not None:
-            dataset.attrs.update({k: v for k, v in header_attrs.items() if v})
-
-        dataset.attrs['history'] = ("Created by pytroll/satpy on " +
-                                    str(datetime.utcnow()))
-        dataset.attrs['conventions'] = 'CF-1.7'
-        engine = kwargs.pop("engine", 'h5netcdf')
-        for key in list(kwargs.keys()):
-            if key not in ['mode', 'format', 'group', 'encoding', 'unlimited_dims', 'compute']:
-                kwargs.pop(key, None)
-        return dataset.to_netcdf(filename, engine=engine, **kwargs)
+            root.attrs.update({k: v for k, v in header_attrs.items() if v})
+        if groups is None:
+            # Groups are not CF-1.7 compliant
+            root.attrs['Conventions'] = 'CF-1.7'
```

Comment:
Could/should we put the draft version of the next CF standard or allow the user to specify the Convention but default to 1.7?

Reply:
I think we should take this in another PR. This one is already large enough. I'm merging this now.